src/Deserialization/IdFactory.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Symbolica.Deserialization
+{
+    internal sealed class IdFactory : IIdFactory
+    {
+        private readonly Dictionary<IntPtr, ulong> _ids;
```

Comment:
Do we care about thread safety here?

Reply:
No, deserialization of a module is single-threaded and a single `IdFactory` is used for a single deserialization.